github-pullrequest-plugin/src/test/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverterTest.java

```diff
@@ -116,4 +126,32 @@ public void shouldReturnCauseOnWhen1OpenedWeGetSecondOneAndHaveEvents() throws E
         assertThat("reason in cause", cause.getReason(), containsString("open"));
         assertThat("pr num in cause", cause.getNumber(), is(2));
     }
+
+    /**
+     * Test trigger configuration of:
+     *
+     *     1.) Skip PR if label is not present (when label is not present)
+     *     2.) Cause PR if commit changed (when commit has changed)
+     *
+     * Expected result is that the PR should be skipped. No causes should be
+     * identified.
+     */
+    @Test
+    public void shouldSkipEvenThoughNewCommit() throws Exception {
+        GHCommitPointer commitPtr = mock(GHCommitPointer.class);
+        when(local.getPulls()).thenReturn(ImmutableMap.of(3, localPR));
+        when(localPR.getHeadSha()).thenReturn("this is not the sha you are looking for");
+        when(remotePR.getNumber()).thenReturn(3);
+        when(remotePR.getState()).thenReturn(GHIssueState.OPEN);
+        when(remotePR.getHead()).thenReturn(commitPtr);
+        when(trigger.getEvents()).thenReturn(asList(
+                new GitHubPRLabelNotExistsEvent(new GitHubPRLabel("notfound"), true),
+                new GitHubPRCommitEvent()));
+
+        List<GitHubPRCause> causes = from(ImmutableSet.of(remotePR))
+                .transform(toGitHubPRCause(local, tlRule.getListener(), trigger))
+                .filter(notNull()).toList();
+
+        assertThat("no cause", causes, hasSize(0));
```

Comment:
mm.. Cause should be returned by event, but it should have skip=true.  
from().transform(). should contain some filterSkip... logic. and only then cause should be zero. Because this list of causes is used for triggering...
The initial logic was in one body and @lanwen split it. Please try debug root triggering logic in Trigger.